src/games/strategy/triplea/ui/screen/IDrawable.java

```diff
@@ -128,26 +128,29 @@ public void draw(final Rectangle bounds, final GameData data, final Graphics2D g
         return;
       }
     }
-    final Rectangle territoryBounds = mapData.getBoundingRect(territory);
+
+    // get font metrics
     graphics.setFont(MapImage.getPropertyMapFont());
     graphics.setColor(MapImage.getPropertyTerritoryNameAndPUAndCommentcolor());
     final FontMetrics fm = graphics.getFontMetrics();
+
+    // if we specify a placement point, use it otherwise try to center it
     int x;
     int y;
-    // if we specify a placement point, use it
-    // otherwise, put it in the center
     final Point namePlace = mapData.getNamePlacementPoint(territory);
     if (namePlace == null) {
+      final Rectangle territoryBounds = getBestTerritoryNameRect(mapData, territory, fm);
       x = territoryBounds.x;
       y = territoryBounds.y;
-      x += (int) territoryBounds.getWidth() >> 1;
-      y += (int) territoryBounds.getHeight() >> 1;
-      x -= fm.stringWidth(territory.getName()) >> 1;
-      y += fm.getAscent() >> 1;
+      x += (int) territoryBounds.getWidth() / 2;
```

Comment:
Why do this increment of `x` on abrand new line? Pls just inline it with line 143.